react/components/SearchResult.js

```diff
@@ -101,6 +110,113 @@ class SearchResult extends Component {
         showLoadingAsOverlay: true,
       })
     }
+
+    document.addEventListener('scroll', this.handleScroll)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('scroll', this.handleScroll)
+  }
+
+  handleScroll = () => {
+    const scroll = this.props.leanMode ? Infinity : window.scrollY
+    const {runtime: { hints: { mobile } }} = this.props
+    
+    if (typeof scroll !== 'number' || !mobile) return
+
+    if (scroll > this.state.scrollValue) this.handleScrollDown()
+    else if (scroll < this.state.scrollValue) this.handleScrollUp(scroll)
+
+    this.setState({scrollValue: scroll})
+  }
+
+  handleScrollUp = scrollValue => {
+    const marginValue = scrollValue * CONVERSION_MARGIN_VALUE
+
+    const searchOptionsBarElement = this.searchOptionsBar.current
+    if (searchOptionsBarElement) {
+      searchOptionsBarElement.style.transition = `opacity ${FADE_DURATION} ${FADE_DELAY}`
```

Comment:
You should apply this transition to `WebkitTransition` too in order support older browsers, so you should have these two lines:
```suggestion
searchOptionsBarElement.style.transition = `opacity ${FADE_DURATION} ${FADE_DELAY}`
searchOptionsBarElement.style.WebkitTransition = `opacity ${FADE_DURATION} ${FADE_DELAY}`
```

Reply:
I have changed the direct style application for a class application based on the state change. The animations are from animate.css library and I think it covers this case.